jaraco/windows/clipboard.py

```diff
@@ -82,42 +80,73 @@ class HTMLSnippet(object):
     <https://docs.microsoft.com/en-us/windows/win32/dataxchg/html-clipboard-format>`_.
     """
 
+    # The non-normative grammar defines markers with a space before "-->",
+    # but the body text defines them without spaces and states "with no
+    # whitespace chars within each comment itself".
+    # Actually, Edge and Firefox don't include spaces, so we also don't include
+    # a space.
+    START_FRAGMENT_MARKER = b'<!--StartFragment-->'
+    END_FRAGMENT_MARKER = b'<!--EndFragment-->'
+
     def __init__(self, handle):
-        self.data = nts(raw_data(handle).decode('utf-8'))
+        self.data = raw_data(handle)
         self.headers = self.parse_headers(self.data)
 
     @property
     def html(self):
-        return self.data[self.headers['StartHTML'] :]
+        start_html = self.headers['StartHTML']
+        end_html = self.headers['EndHTML']
+        if start_html == -1:
+            start_html = self.headers['StartFragment'] - len(
+                HTMLSnippet.START_FRAGMENT_MARKER
+            )
+        if end_html == -1:
+            end_html = self.headers['EndFragment'] + len(
+                HTMLSnippet.END_FRAGMENT_MARKER
+            )
+        return self.data[start_html:end_html].decode('utf-8')
 
     @property
     def fragment(self):
-        return self.data[self.headers['StartFragment'] : self.headers['EndFragment']]
+        start_fragment = self.headers['StartFragment']
+        end_fragment = self.headers['EndFragment']
+        return self.data[start_fragment:end_fragment].decode('utf-8')
+
+    @property
+    def selection(self):
+        start_selection = (
+            self.headers.get('StartSelection') or self.headers['StartFragment']
+        )
+        end_selection = self.headers.get('EndSelection') or self.headers['EndFragment']
+        return self.data[start_selection:end_selection].decode('utf-8')
 
     @staticmethod
     def parse_headers(data):
-        d = io.StringIO(data)
-
-        def header_line(line):
-            return re.match(r'(\w+):(.*)', line)
-
-        headers = map(header_line, d)
-        # grab headers until they no longer match
-        headers = itertools.takewhile(bool, headers)
-
         def best_type(value):
             try:
                 return int(value)
             except ValueError:
-                pass
-            try:
-                return float(value)
-            except ValueError:
-                pass
```

Comment:
It looks like we lost support for floats. Maybe it's not needed, but it feels like such a change should be made explicitly and separately and not as part of a change for multibyte support.